docs/next/modules/en/pages/user/gitops.adoc

```diff
@@ -0,0 +1,188 @@
+
+A well-structured repository simplifies management and improves clarity. The following folder separation strategy is recommended:
+
+----
+repo-root/
+│── fleet/
+│   ├── providers/         # CAPIProvider files to use with templates
+│   │   ├── core/          # Core provider configuration
+│   │   ├── infra-docker/  # Docker provider configuration
+│   │   ├── infra-aws/     # AWS provider configuration
+│   ├── clusters/          # Cluster-specific configurations
+│   │   ├── prod/          # Production clusters
+│   │   ├── staging/       # Staging clusters
+│   │   ├── dev/           # Development clusters
+│   ├── addons/            # Application-specific manifests focused on cluster bootstrapping, like CNI, CCM, CPI configurations.
+│   │   ├── cni/           # Group applications based on purpose. Provide fleet.yaml per each sub-directtory to maintain bundle separation.
+│   │   ├── ccm/
+│   ├── templates/         # CAPI ClusterClass templates for cluster provisioning
+│   ├── fleet.yaml         # Fleet bundle configuration
+│── overlays/              # Kustomize overlays for different environments
+│   ├── prod/.             # Kustomize overlays for production environments
+│   ├── staging/.          # Kustomize overlays for staging environments
+│   ├── dev/.              # Kustomize overlays for development environments
+----
+
+== Providers configuration
+
+CAPI providers define the infrastructure components necessary for cluster provisioning. A simple setup may look like:
+
+----
+repo-root/
+│── fleet/
+│   ├── providers/
+│   │   ├── core/
+│   │   |   ├── core.yaml  # Core `CAPIProvider` definition
+│   │   |   ├── fleet.yaml # Fleet.yaml definition for the provider setup
+----
+
+where the core provider contains:
+
+```yaml
+apiVersion: turtles-capi.cattle.io/v1alpha1
+kind: CAPIProvider
+metadata:
+  name: cluster-api
+spec:
+  type: core
+```
+
+== Templates configuration
+
+Specify `dependsOn` in `fleet.yaml` for specific `CAPIProvider` bundles or external `GitRepo` components. This ensures templates roll out in the correct order and prevents issues where component controllers are missing or not ready to accept `ClusterClass` definitions.
+
+```yaml
+namespace: capa-clusters
+dependsOn:
+  - name: core      # Ensure core bundle is installed first
+  - name: infra-aws # Wait for AWS provider installation
+```
+
+== Clusters configuration
+
+Use labels to dynamically match clusters with required addons, allowing to deploy only nessesary workloads.
+
+Example label usage:
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  name: aws-cluster
+  labels:
+    cni: calico
+    ccm: aws
+    env: dev
+```
+
+[WARNING]
+====
+It is recommended to separate cluster definitions from the templates, as such cluster bundle removal would allow CAPI controller to process deletion in the correct order. Removing `Clusters` and `ClusterClasses` simultaniously may yield unexpected outcomes.
```

Comment:
```suggestion
It is recommended to separate cluster definitions from the templates, as such cluster bundle removal would allow CAPI controller to process deletion in the correct order. Removing `Clusters` and `ClusterClasses` simultaneously may yield unexpected outcomes.
```